.editorconfig

```diff
@@ -176,18 +176,306 @@ end_of_line = crlf
 # Code files
 [*.{cs,vb}]
 
-dotnet_diagnostic.UseAsyncSuffix.severity = error
+# Async Threading Rules
 
-dotnet_diagnostic.UseConfigureAwait.severity = error
+# Use Async suffix for async methods
+dotnet_diagnostic.VSTHRD200.severity = error
 
-dotnet_diagnostic.AvoidAsyncSuffix.severity = none
+# Use .ConfigureAwait(bool)
+dotnet_diagnostic.VSTHRD111.severity = error
 
-dotnet_diagnostic.AvoidAsyncVoid.severity = none
+# Avoid async void methods
+dotnet_diagnostic.VSTHRD100.severity = warning
 
+# Critical Threading Errors
+
+# Avoid legacy thread switching methods
+dotnet_diagnostic.VSTHRD001.severity = warning
+
+# Avoid problematic synchronous waits
+dotnet_diagnostic.VSTHRD002.severity = warning
+
+# Avoid awaiting foreign Tasks
+dotnet_diagnostic.VSTHRD003.severity = warning
+
+# Await SwitchToMainThreadAsync
+dotnet_diagnostic.VSTHRD004.severity = error
+
+# Invoke single-threaded types on Main thread
+dotnet_diagnostic.VSTHRD010.severity = warning
+
+# Use AsyncLazy<T>
+dotnet_diagnostic.VSTHRD011.severity = none #error requires the use of Microsoft.VisualStudio.Threading to get access to the AsyncLazy<T> type
```

Comment:
inorder to enable this we need to use Microsoft.VisualStudio.Threading but the package is not compatable with our .net 45 targets. same with some other older targets.

@bgavrilMS

Reply:
Yeah, bummer.